src/components/login/GithubLoginButton.tsx

```diff
@@ -0,0 +1,54 @@
+import { type Provider } from '@supabase/supabase-js'
+import { useTransition } from 'react'
+import { Button } from '../ui/button'
+import { loginAction } from '@/api/auth'
+import { useRouter } from 'next/navigation'
+import { toast } from '../ui/use-toast'
+import { ButtonLoadingSpinner } from '../ui/loading'
+import Image from 'next/image'
+
+export default function GithubLoginButton() {
+  const [isPending, startTransaction] = useTransition()
+  const router = useRouter()
+
+  const handleClickLoginButton = (provider: Provider) => {
+    startTransaction(async () => {
+      const { errorMessage, url } = await loginAction(provider)
+      if (!errorMessage && url) {
+        router.push(url)
```

Comment:
Really small change, but could we also add a toast for logging in (it will only popup for a couple seconds (until the page is re-routed), but it might be nice to include more feedback!

```typescript
toast({
  title: 'Success!',
  description: 'Signed in succesfully',
})
```